scripts/scripts.js

```diff
@@ -819,165 +452,310 @@ export function addFavIcon(href) {
- * @param {path} path to *.metadata.json
- * @returns {Object} containing sanitized meta data
+ * Decorates the page.
  */
-
-async function getMetadataJson(path) {
-  const resp = await fetch(path.split('.')[0]);
-  const text = await resp.text();
-  const meta = {};
-  if (resp.status === 200 && text && text.includes('<head>')) {
-    const headStr = text.split('<head>')[1].split('</head>')[0];
-    const head = document.createElement('head');
-    head.innerHTML = headStr;
-    const metaTags = head.querySelectorAll(':scope > meta');
-    metaTags.forEach((metaTag) => {
-      const name = metaTag.getAttribute('name') || metaTag.getAttribute('property');
-      const value = metaTag.getAttribute('content');
-      if (meta[name]) {
-        meta[name] += `, ${value}`;
-      } else {
-        meta[name] = value;
-      }
-    });
-  }
-  return (JSON.stringify(meta));
+async function loadPage(doc) {
+  // eslint-disable-next-line no-use-before-define
+  await loadEager(doc);
+  // eslint-disable-next-line no-use-before-define
+  await loadLazy(doc);
+  // eslint-disable-next-line no-use-before-define
+  loadDelayed(doc);
 }
 
-/**
- * gets a blog article index information by path.
- * @param {string} path indentifies article
- * @returns {object} article object
- */
+export function initHlx() {
+  window.hlx = window.hlx || {};
+  window.hlx.lighthouse = new URLSearchParams(window.location.search).get('lighthouse') === 'on';
+  window.hlx.codeBasePath = '';
 
-export async function getBlogArticle(path) {
-  const json = await getMetadataJson(`${path}.metadata.json`);
-  const meta = JSON.parse(json);
-  if (meta['og:title']) {
-    const articleMeta = {
-      description: meta.description,
-      title: meta['og:title'],
-      image: meta['og:image'],
-      imageAlt: meta['og:image:alt'],
-      date: meta['publication-date'],
-      path,
-      category: meta.category,
-    };
-    return (articleMeta);
+  const scriptEl = document.querySelector('script[src$="/scripts/scripts.js"]');
+  if (scriptEl) {
+    try {
+      [window.hlx.codeBasePath] = new URL(scriptEl.src).pathname.split('/scripts/scripts.js');
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.log(e);
+    }
   }
-  return null;
 }
 
-/**
- * fetches the string variables.
- * @returns {object} localized variables
+initHlx();
+
+/*
+ * ------------------------------------------------------------
+ * Edit above at your own risk
+ * ------------------------------------------------------------
  */
 
-export async function fetchPlaceholders() {
-  if (!window.placeholders) {
-    const resp = await fetch(`${getRootPath()}/placeholders.json`);
-    const json = await resp.json();
-    window.placeholders = {};
-    json.data.forEach((placeholder) => {
-      window.placeholders[placeholder.Key] = placeholder.Text;
+const RUM_GENERATION = 'project-1'; // add your RUM generation information here
+const PRODUCTION_DOMAINS = [];
+
```

Comment:
LCP_BLOCKS would probably have to be added here, with the lcp blocks from https://github.com/adobe/business-website/pull/199/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cL1102

Reply:
Even if waitForLCP() isn't used, do you still want to set LCP_BLOCKS?

Reply:
...waitForLCP has to be used. it blocks the display and other network activity before the LCP happened.